### PR TITLE
Add button to generate new card from withing room

### LINKED ISSUE #3: Add "Lobby" functionality to let players start a game after joining a room
The rush to join a room after the seed is revealed can be stressful and error-prone.

This can be alleviated by letting players join a room before committing to a seed. The implementation could either allow delayed setting of the seed or "re-setting" of the seed after a room is already created (likely creating a new game).

### PR DESCRIPTION
**Pull request by FelixNemis**
Fixes #3 
The button brings up a modal dialog with settings for a new card.
The "hide card initially" setting updates the room since that's where that setting is saved.
I think custom json boards are working too, the input should be hidden when you don't select custom from the dropdown.

In the process I cleaned up the way javascript accesses the room settings a bit.
I put all the room settings into an object called ROOM_SETTINGS
the /room_settings view returns a json object to save to it, as well as html to update the room settings panel

### CODE REVIEW

File: bingosync-app/templates/bingosync/bingosync.html [hunk 1]
```diff
@@ -88,7 +93,30 @@ <h4 style="display: inline-block; position: absolute; right: 0; bottom: 0;">
 
         // hook up chat contents toggling
         initializeChatSettings($("#chat-settings"), $("#bingo-chat"));
+
+        // collapse event for the room settings panel
+        // I put it here becuase the template is loaded each time a new card is created
+        $("#room-settings-collapse").on("mousedown", function() {
```
Comment (kbuzsaki) [inline bingosync-app/templates/bingosync/bingosync.html]: Minimizing room settings doesn't seem to work any more after I generate a new card. Maybe this `.on("mousedown", ...)` should go in refreshSettingsPanel?

File: bingosync-app/bingosync/models.py [hunk 2]
```diff
@@ -601,6 +601,7 @@ def get_all_for_room(room):
         color_events = list(ColorEvent.objects.filter(player__room=room))
         revealed_events = list(RevealedEvent.objects.filter(player__room=room))
         connection_events = list(ConnectionEvent.objects.filter(player__room=room))
+        new_card_events = list(NewCardEvent.objects.filter(player__room=room))
         all_events = chat_events + goal_events + color_events + revealed_events + connection_events
```
[the remark this answers is not in the document]
Comment (FelixNemis) [reply bingosync-app/bingosync/models.py]: ah I was wondering about that, though I forgot at some point.

File: bingosync-app/bingosync/models.py [hunk 3]
```diff
@@ -621,6 +622,16 @@ def to_json(self):
             "timestamp": self.json_timestamp
         }
 
+class NewCardEvent(Event):
```
[the remark this answers is not in the document]
Comment (FelixNemis) [reply bingosync-app/bingosync/models.py]: Good idea

File: bingosync-app/bingosync/views.py [hunk 4]
```diff
@@ -71,6 +81,60 @@ def room_board(request, encoded_room_uuid):
     board = room.current_game.board
     return JsonResponse(board, safe=False)
 
+# AJAX view to render the room settings panel
+def room_settings(request, encoded_room_uuid):
+    if not request.is_ajax():
+        return HttpResponseBadRequest("This view accepts only AJAX requests")
```
[the remark this answers is not in the document]
Comment (FelixNemis) [reply bingosync-app/bingosync/views.py]: I guess not really

File: bingosync-app/templates/bingosync/bingosync.html (hunk 1 above)
Comment (FelixNemis) [reply bingosync-app/templates/bingosync/bingosync.html]: good catch

File: bingosync-app/templates/bingosync/bingosync.html [hunk 5]
```diff
@@ -54,9 +52,18 @@ <h4 style="display: inline-block; position: absolute; right: 0; bottom: 0;">
         </div>
     </div>
 </div>
+{% include 'bingosync/new_card_dialog.html' %}
 <script src="{% static "bingosync/bingosync.js" %}"></script>
 <script>
     $(document).ready(function() {
+        // populate initial settings
+        ROOM_SETTINGS = {
```
[the remark this answers is not in the document]
Comment (FelixNemis) [reply bingosync-app/templates/bingosync/bingosync.html]: how's this?

File: bingosync-app/bingosync/forms.py [hunk 6]
```diff
@@ -162,3 +162,16 @@ def clean(self):
     def get_goal_list(self):
         return self.json_str
 
+# just for generating html form for the modal popup
+class NewCardForm(forms.Form):
+    game_type = forms.ChoiceField(label="Game", choices=GameType.form_choices())
```
[the remark this answers is not in the document]
Comment (FelixNemis) [reply bingosync-app/bingosync/forms.py]: done

File: bingosync-app/bingosync/models.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (FelixNemis) [reply bingosync-app/bingosync/models.py]: done

**Pull request merged**